Check-in PSA related auto-generated files to support online compiler.

jainvikas8 opened the pull request:
This PR is to add support for an online compiler which is currently failing LPC55S69 (PSA target).

Support the online compiler build for PSA targets, some files are auto-generated (LPC55S69_NS and ARM_MUSCA_A1_NS) by the script from the offline build system. Since these files are identical for these V8-M targets it would be good to check them in to support online compiler as they are COMPONENT_SPE and services related.
 
Offline build system only for PSA related targets will be affected, as the files are checked in, it will use those files instead of auto-generating them.

jainvikas8 committed:
Add PSA auto-generated files to components

To support online compiler build for PSA targets, some files are
auto-generated (LPC55S69_NS and ARM_MUSCA_A1_NS) by the script
from offline build system. Since these files are identical for these
V8-M targets it would be good to check them in to support online
compiler as they are COMPONENT_SPE and services related.

In folder: components/TARGET_PSA/
  --TARGET_MBED_SPM
    --COMPONENT_SPE
      psa_setup.c
  --TARGET_TFM
    --COMPONENT_SPE
      --inc
	tfm_partition_defs.inc
        tfm_partition_list.inc
        tfm_service_list.inc
        tfm_spm_signal_defs.h
  --services
    --inc
      autogen_sid.h
      mbed_spm_partitions.h

Signed-off-by: Vikas Katariya <Vikas.Katariya@arm.com>

mark-edgeworth reviewed (changes requested):
This will change both the online and offline (mbed-cli) builds. It looks like the change removes all of the auto generation in favor of checked-in files for both online and offline building (ie the generator will not now be used at all). Is this intentional. If so, can you add a note to the PR description to indicate that offline builds are affected too?

urutva commented on tools/build.py:
```diff
@@ -44,7 +44,6 @@
 from tools.utils import NoValidToolchainException
 from tools.utils import print_end_warnings
-from tools.psa import generate_psa_sources, clean_psa_autogen
 from tools.resources import OsAndSpeResourceFilter
```
@jainvikas8 Do we still need `OsAndSpeResourceFilter `?

jainvikas8 replied:
I'm not sure about this, as the main objective is to not make drastic changes to the offline build system and only remove auto-generation of header files.

urutva commented on tools/build.py:
```diff
@@ -191,12 +187,6 @@ def main():
-                            )
-
                         resource_filter = None
                         if target.is_PSA_secure_target:
```
PSA secure target is built by `tools/psa/release.py`. Can we remove this?

jainvikas8 replied:
Same comment as https://github.com/ARMmbed/mbed-os/pull/11685#discussion_r343076062

urutva commented on tools/make.py:
```diff
@@ -339,12 +335,6 @@ def main():
-                )
-
             resource_filter = None
             if target.is_PSA_secure_target:
```
PSA secure target is built by `tools/psa/release.py` right. Can we remove this?

jainvikas8 replied:
Same comment as https://github.com/ARMmbed/mbed-os/pull/11685#discussion_r343076062

urutva commented on tools/project.py:
```diff
@@ -391,11 +389,6 @@ def main():
-                )
-
             resource_filter = None
             if target.is_PSA_secure_target:
```
PSA secure target is built by `tools/psa/release.py` right. Can we remove this?

jainvikas8 replied:
Same comment as https://github.com/ARMmbed/mbed-os/pull/11685#discussion_r343076062

urutva commented on tools/test.py:
```diff
@@ -244,12 +240,6 @@ def main():
                 if target.is_PSA_secure_target:
```
Will `mbed test` be run on secure targets? If not then can we remove this?

jainvikas8 replied:
https://github.com/ARMmbed/mbed-os/pull/11685#discussion_r343076062

jainvikas8 committed:
Remove auto-generation for non-secure PSA targets.

Only generate PSA headers/source related to components and services
when Secure build is initiated during compile time of PSA targets.
Let the Non-secure build rely on the checked-in files already present.

Signed-off-by: Vikas Katariya <Vikas.Katariya@arm.com>

Patater commented on components/TARGET_PSA/services/inc/mbed_spm_partitions.h:
```diff
@@ -0,0 +1,307 @@
+/* Copyright (c) 2017-2019 ARM Limited
```
Is this useful to TF-M builds? Should we place it under `components/TARGET_PSA/TARGET_MBED_SPM/inc/`?

Patater replied:
Looks Mbed services use this, even when TF-M SPM is used. OK

Patater commented on components/TARGET_PSA/services/inc/autogen_sid.h:
```diff
@@ -0,0 +1,110 @@
+/* Copyright (c) 2019 ARM Limited
```
Is this used by TFM? If not, consider putting into `components/TARGET_PSA/TARGET_MBED_SPM/services/inc/`.

Patater replied:
This is used even when TF-M SPM is used, so OK.

Patater reviewed (approved):
LGTM

jainvikas8 committed:
PSA release script changes.

Since the offline build is made to auto-generate PSA related components
and services for Secure targets, we can change the output directory to
update the files in the respective locations.
TARGET_PSA
  --TARGET_MBED_SPM
    --COMPONENT_SPE
      psa_setup.c
  --TARGET_TFM
    --COMPONENT_SPE
      --inc
	tfm_partition_defs.inc
        tfm_partition_list.inc
        tfm_service_list.inc
        tfm_spm_signal_defs.h
  --services
    --inc
      autogen_sid.h
      mbed_spm_partitions.h

The release script is been modified to commit these files if there are
any changes detected when `--commit` argument is passed.
Cleaning of auto-generated is been removed as it uses the main directory
for its operations, but PSA auto-generation will work if any of the
service and application-based manifests are updated.

Signed-off-by: Vikas Katariya <Vikas.Katariya@arm.com>

jainvikas8 committed:
Documentation update for tools/psa

Signed-off-by: Vikas Katariya <Vikas.Katariya@arm.com>
